### PR TITLE
Add codecov.yml file

### LINKED ISSUE #214: Add yaml configuration for CodeCov
## Description
We currently support codecov, since it requires a YAML configuration file more about this YAML configuration can be found [here](https://docs.codecov.io/docs/codecov-yaml).
For any more clarification/question ping @gmuraru or me on slack.

### LINKED ISSUE #267: CodeCov - show on PR
## Description
CodeCov shows the code coverage for the tests when we do a merge, but ideally, we would want to have a comment on the PR.
Also, codecov should only be run, but without failing (have the red X) on the main branch.

## Are you interested in working on this improvement yourself?
- Yes, I am.

### PR DESCRIPTION
**Pull request by shubhank-saxena**
## Description
This PR fixes #214 and fixes #267
This PR adds codecov settings file
@aanurraj I have added basic configuration, please let me know if there are any further configurations needed as per the requirements!

## Affected Dependencies
NA

## How has this been tested?
NA

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [X] My changes are covered by tests

### CODE REVIEW

**Comment by rasswanth-s** [comment]
Hi @shubhank-saxena ,
I think adding `codecov.yml` also solves this issue #267 ,[Link](https://about.codecov.io/blog/python-code-coverage-using-github-actions-and-codecov/) .  could do you simple change (in src or test files ,adding a new line or some minor change) so that the CI could run , and we could validate if `Codecov comments` on pull requests.

**Comment by shubhank-saxena** [comment]
> Hi @shubhank-saxena ,
> I think adding `codecov.yml` also solves this issue #267 ,[Link](https://about.codecov.io/blog/python-code-coverage-using-github-actions-and-codecov/) . could do you simple change (in src or test files ,adding a new line or some minor change) so that the CI could run , and we could validate if `Codecov comments` on pull requests.

Sure. Will do that right away!

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+comment:
+  layout: "reach,diff,flags,files,footer"
+  behavior: default
+  require_changes: no
+
+  coverage:
+  precision: 2
+  round: down
+  range: "70...100"
```
Comment (aanurraj) [inline .codecov.yml:12]: can you make it to above 90 % ?

**Review by aanurraj — CHANGES_REQUESTED** [review]

File: .codecov.yml (hunk 1 above)
Comment (shubhank-saxena) [reply .codecov.yml:12]: Done!

**Comment by rasswanth-s** [comment]
@shubhank-saxena , I added a minor change to trigger the CI, we could go for merge, if codecov comments on PR after CI.

**Review by aanurraj — APPROVED** [review]
Looks good, lets see if it solves or issue :D

**Pull request merged**